flink-table/flink-table-api-java/src/main/java/org/apache/flink/table/api/config/OptimizerConfigOptions.java

```diff
@@ -56,33 +56,33 @@
 							"when there is data skew in distinct aggregation and gives the ability to scale-up the job. " +
 							"Default is false.");
 
-	public static final ConfigOption<Integer> SQL_OPTIMIZER_DISTINCT_AGG_SPLIT_BUCKET_NUM =
-			key("sql.optimizer.distinct-agg.split.bucket-num")
+	public static final ConfigOption<Integer> TABLE_OPTIMIZER_DISTINCT_AGG_SPLIT_BUCKET_NUM =
+			key("table.optimizer.distinct-agg.split.bucket-num")
 					.defaultValue(1024)
 					.withDescription("Configure the number of buckets when splitting distinct aggregation. " +
 							"The number is used in the first level aggregation to calculate a bucket key " +
 							"'hash_code(distinct_key) % BUCKET_NUM' which is used as an additional group key after splitting.");
 
-	public static final ConfigOption<Boolean> SQL_OPTIMIZER_REUSE_SUB_PLAN_ENABLED =
-			key("sql.optimizer.reuse.sub-plan.enabled")
+	public static final ConfigOption<Boolean> TABLE_OPTIMIZER_REUSE_SUB_PLAN_ENABLED =
+			key("table.optimizer.reuse.sub-plan.enabled")
 					.defaultValue(true)
 					.withDescription("When it is true, optimizer will try to find out duplicated " +
 							"sub-plan and reuse them.");
 
-	public static final ConfigOption<Boolean> SQL_OPTIMIZER_REUSE_TABLE_SOURCE_ENABLED =
-			key("sql.optimizer.reuse.table-source.enabled")
+	public static final ConfigOption<Boolean> TABLE_OPTIMIZER_REUSE_SOURCE_ENABLED =
+			key("table.optimizer.reuse.source.enabled")
 					.defaultValue(true)
-					.withDescription("When it is true, optimizer will try to find out duplicated table-source and " +
-							"reuse them. This works only when " + SQL_OPTIMIZER_REUSE_SUB_PLAN_ENABLED.key() + " is true.");
+					.withDescription("When it is true, optimizer will try to find out duplicated table source and " +
+							"reuse them. This works only when " + TABLE_OPTIMIZER_REUSE_SUB_PLAN_ENABLED.key() + " is true.");
 
-	public static final ConfigOption<Boolean> SQL_OPTIMIZER_PREDICATE_PUSHDOWN_ENABLED =
-			key("sql.optimizer.predicate-pushdown.enabled")
+	public static final ConfigOption<Boolean> TABLE_OPTIMIZER_PREDICATE_PUSHDOWN_ENABLED =
+			key("table.optimizer.predicate-pushdown.enabled")
 					.defaultValue(true)
 					.withDescription("If it is true, enable predicate pushdown to the FilterableTableSource. " +
 			"Default value is true.");
 
-	public static final ConfigOption<Boolean> SQL_OPTIMIZER_JOIN_REORDER_ENABLED =
-			key("sql.optimizer.join-reorder.enabled")
+	public static final ConfigOption<Boolean> TABLE_OPTIMIZER_JOIN_REORDER_ENABLED =
+			key("table.optimizer.join-reorder.enabled")
 					.defaultValue(false)
 					.withDescription("Enables join reorder in optimizer cbo. Default is disabled.");
```

Comment:
what is "cbo"?

Reply:
cost based optimizer (e.g. VolcanoPlanner in Calcite)
another concept is "rbo", means rule based optimizer (e.g. HepPlanner in Calcite)

Reply:
@godfreyhe , is it only enabled in CBO? I mean, users shouldn't care about what optimizer is used in Flink.